cpp/build-support/run-benchmark.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eux
+
+# See https://github.com/google/benchmark/blob/6a5c379ca/tools/gbench/report.py#L39
+# https://github.com/scipy/scipy/blob/c3fa90dcfcaef71/scipy/stats/stats.py#L4957
+: ${BENCHMARK_REPETITIONS:=20}
+
+PWD=$(cd $(dirname $BASH_SOURCE); pwd)
+
+convert_benchmark() {
+  local version=$1
+
+  ${PWD}/convert-benchmark.py ${version}
+}
+
+run_benchmark() {
+  local bin=$1
+  shift
+
+  "${bin}" \
+    --benchmark_format=json \
+    --benchmark_repetitions=${BENCHMARK_REPETITIONS} \
+    $@
```

Comment:
`"$@"` is better for arguments that include a space.

https://linux.die.net/man/1/bash
> When the expansion occurs within double quotes, each parameter expands to a separate word. That is, "$@" is equivalent to "$1" "$2" ... 

```bash
#!/bin/bash

python -c 'import sys; print(sys.argv)' $@
python -c 'import sys; print(sys.argv)' "$@"
```

```console
% /tmp/a.sh 1 "2 3" 4
['-c', '1', '2', '3', '4']
['-c', '1', '2 3', '4']
```

Reply:
There's always something to learn! (I should have use shellcheck).

Reply:
https://github.com/apache/arrow/pull/4118